src/ImageSharp/Formats/Jpeg/Port/Components/FrameComponent.cs

```diff
@@ -0,0 +1,74 @@
+﻿// <copyright file="FrameComponent.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp.Formats.Jpeg.Port.Components
+{
+    using System;
+
+    using ImageSharp.Memory;
+
+    /// <summary>
+    /// Represents a single frame component
+    /// </summary>
+    internal struct FrameComponent : IDisposable
+    {
+        /// <summary>
+        /// Gets or sets the component Id
+        /// </summary>
+        public byte Id;
+
+        /// <summary>
+        /// TODO: What does pred stand for?
+        /// </summary>
+        public int Pred;
+
+        /// <summary>
+        /// Gets or sets the horizontal sampling factor.
+        /// </summary>
+        public int HorizontalFactor;
+
+        /// <summary>
+        /// Gets or sets the vertical sampling factor.
+        /// </summary>
+        public int VerticalFactor;
+
+        /// <summary>
+        /// Gets or sets the identifier
+        /// </summary>
+        public byte QuantizationIdentifier;
+
+        /// <summary>
+        /// Gets or sets the block data
+        /// </summary>
+        public Buffer<short> BlockData;
```

Comment:
It's really worth to introduce `Block8x8Int` and refactor this into `Buffer<Block8x8Int>`.

A quick candidate implementation utilizing [Fixed-size buffers](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/unsafe-code-pointers/fixed-size-buffers) and having an `.AsSpan()` method:
```C#
internal unsafe struct Block8x8Int
{
    private fixed short data[64];

    public Span<short> AsSpan() => new Span<short>(this.data, 64);

    public ref short this[int idx] => ref Unsafe.Add(ref Unsafe.As<Block8x8Int, short>(ref this), idx);
}

```